pkg/containermanager/docker/docker_manager.go

```diff
@@ -94,6 +98,59 @@ func (dc dockerContainerManager) Deprovision(ctx context.Context, instanceID str
 	return nil
 }
 
+func (dc dockerContainerManager) Bind(ctx context.Context, bindingConfig containermanager.BindConfig) (*containermanager.BindResponse, error) {
+	containerInfo, err := dc.client.ContainerInspect(ctx, bindingConfig.InstanceId)
+	if err != nil {
+		return nil, err
+	}
+
+	bindings := make(map[string][]containermanager.Binding)
+	for port, dockerBindings := range containerInfo.NetworkSettings.NetworkSettingsBase.Ports {
+		for _, dockerBinding := range dockerBindings {
+			bindings[port.Port()] = []containermanager.Binding{
+				containermanager.Binding{
+					HostIP: dockerBinding.HostIP,
+					Port:   dockerBinding.HostPort,
+				},
+			}
+		}
+	}
+
+	file, err := ioutil.TempFile("", uuid.New())
+	if err != nil {
+		return nil, err
+	}
+	downloadFile(file.Name(), bindingConfig.Params.ContractUrl)
+	defer os.RemoveAll(file.Name())
```

Comment:
this and the previous five lines or so should move into the generic part of bind. It's not docker specific and needs to be reused by any other containermanagers.